src/controls/Controls.spec.js

```diff
@@ -12,9 +12,50 @@ describe('<Controls />', () => {
 
     expect(tree).toMatchSnapshot();
   });
-    it('button should lock when clicked', () => {
-    const { getByText } = render(<Controls locked={ true } closed={ false }/>);
-    expect(getByText(/lock gate/i));
-    expect(getByText(/close gate/i));
+    it('Opened Gate', () => {
+        const { getByText } = render(<Controls/>);
+            expect(getByText(/lock gate/i));
+            expect(getByText(/close gate/i));
+     })
+    it('Closed Gate', () => {
+        const { getByText } = render(<Controls/>);
+         fireEvent.click(getByText(/Close Gate/i),);
+            expect(getByText(/Lock Gate/i));
+            expect(getByText(/Close Gate/i));
+     })
+    // it('Closed Gate', () => {
+    //     const { getByText } = render(<Controls/>);
+    //       getByText('Close Gate'),
+    //         new MouseEvent('click', {
+    //             locked: false,
+    //             closed: true
+    //         })
+    //         )
+    //      fireEvent.click(getByText(/Close Gate/i),);
+    //         expect(getByText(/Lock Gate/i));
+    //         expect(getByText(/Close Gate/i));
+    //  })
+    it('Lock gate', () => {
+        const { getByText } = render(<Controls/>);
+            fireEvent.click(getByText(/close gate/i));
+             Controls.props.closed = true;
+                expect(getByText(/unlock gate/i));
```

Comment:
the issue here was that the page gets rerendered and doesn't show based off of the previous click